### ISSUE TITLE
Failed setup: Error setting up entry 127.0.0.1 for eufy_security

### ISSUE THREAD

**Issue by gmcdon** [fuatakgun/eufy_security#1274]
**Describe the bug**

When starting integration, receiving the following error:

`Logger: homeassistant.config_entries 
Source: config_entries.py:604
Error setting up entry 127.0.0.1 for eufy_security`

**To reproduce**

1. Start eufy-security-ws (default port: 3000)
3. Install eufy_security
4. Add hub: 127.0.0.1:3000
6. Error message: Failed to set up:
Check the logs

**Expected behavior**

eufy_security to connect to eufy-security-ws

**Additional information**

Go to Settings -> System -> Repairs -> Click on 3 dots (...) -> System Information and get Version and Installation Type to below fields;
- Home Assistant Installation Type (OS - Supvervised - Core): 
- Home Assistant Core Version: 
- Eufy Security Add-on Version: 
- Eufy Security Integration Version:
- Using go2rtc Add-on or webrtc custom integration with integrated go2rtc: 

Hardware Information;

- Debug Logs from Home Assistant (https://github.com/fuatakgun/eufy_security/discussions/624) - Without this, I will automatically resolve this issue

  Error setting up entry 127.0.0.1 for eufy_security

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 604, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/eufy_security/__init__.py", line 56, in async_setup_entry
    await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2108, in async_forward_entry_setups
    await integration.async_get_platforms(platforms)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1177, in async_get_platforms
    import_future.result()
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1177, in async_get_platforms
    import_future.result()
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1177, in async_get_platforms
    import_future.result()
  [Previous line repeated 7 more times]
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1165, in async_get_platforms
    platforms.update(self._load_platforms(platform_names))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1090, in _load_platforms
    platform_name: self._load_platform(platform_name)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1264, in _load_platform
    cache[full_name] = self._import_platform(platform_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1296, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/eufy_security/alarm_control_panel.py", line 6, in <module>
    from homeassistant.components.alarm_control_panel import (
ImportError: cannot import name 'AlarmControlPanelState' from 'homeassistant.components.alarm_control_panel' (/usr/src/homeassistant/homeassistant/components/alarm_control_panel/__init__.py)

**Comment by febalci** [comment]
v8.0.4 works well with HA 2024.9.2. I guess newer versions need HA updated.

**Comment by omar10wahab** [comment]
Somehow updating HA made it work for me as well.

**Comment by danstreeter** [comment]
I'm currently running the following and getting this error. Will try and perform an update of HA over the next few evenings and report my findings:

- Home Assistant Installation Type (OS - Supvervised - Core): Container (Docker)
- Home Assistant Core Version: 2024.10.4
- Eufy Security Add-on Version: 1.9.1
- Eufy Security Integration Version: 8.1.1

**Comment by omar10wahab** [comment]
> @omar10wahab just an FYI, i'm not using a shared account and getting this issue

I can confirm changing to my main account did not make things better. I just realized I had to redownload hacs maybe something to do with the newest release. I'm going to tinker and redownload eufy packages and make sure ha os is up to date. I'm not sure how to read the logs to see if this are breaking but I saw timeout for battery status for my door lock. Idk how that helps me determine what the issue is because there's certain entities that don't populate data but the door lock and door camera devices populate and exist.

**Comment by 1920x1200** [comment]
**TLDR: Check your ports. It has to be 3000 and doesn't like anything else. If you have something else using 3000, you have to change it on that side.**

I just spent hours trying to get this setup and kept running into this error. I finally found the fix! (your millage may vary)

When I would use port 3000 on the add-on side (eufy-security-ws), it would give me errors but if I changed it to something else like 3003, it seemed to connect and find my devices. When I added the Eufy Security "device" and set the port to 3003, it would not like that and fail to connect with the same error mentioned here even thought he ports matched. 

The fix was to find what was using port 3000. In my case it was Z-Wave JS UI. I changed the port for that to 2999 so that port 3000 was now available. I changed the add-on/eufy-security-ws back to 3000 and then added the Eufy Security device and set that to 3000 and now everything works.

**Comment by omar10wahab** [comment]
So I'm not sure if it was related to the port settings which I believe mine were all default but I noticed my doorbell camera starting to show up some data but then I realized the door lock was showing some data but some entities were disabled. I just had to reenable them to get them to work and now everything is back to normal. I have gone through several HA updates and integration updates with no changes made to my eufy account and using default settings with the integration. Idk why the entities became disabled but I have no issues now. Just sharing my experience
